### PR TITLE
Implement Missing Command Line Quoting on Windows

### PR DESCRIPTION
**Pull request by jmittert**

### CODE REVIEW

**Comment by jmittert** [comment]
@compnerd

File: Foundation/Process.swift [hunk 1]
```diff
@@ -318,12 +374,11 @@ open class Process: NSObject {
         }
 
 #if os(Windows)
-        // TODO(compnerd) quote the commandline correctly
-        // https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
         var command: [String] = [launchPath]
         if let arguments = self.arguments {
           command.append(contentsOf: arguments)
         }
+        command = command.map(quoteWindowsCommandArg)
```
Comment (jakepetroules) [inline Foundation/Process.swift]: Can you please keep the transformation of the `[String]` input to the final `String` passed to the native API, localized to a single function call? Like:

```swift
private func quoteWindowsCommandLine(_ commandLine: [String]) -> String {
    func quoteWindowsCommandArg(arg: String) -> String { /* ... */ }
    return commandLine.map(quoteWindowsCommandArg).joined(separator: " ")
}
```

File: Foundation/Process.swift [hunk 2]
```diff
@@ -102,6 +102,62 @@ private func processIsEqual(_ a : UnsafeRawPointer?, _ b : UnsafeRawPointer?) ->
     return true
 }
 
+#if os(Windows)
+private func quoteWindowsCommandArg(arg: String) -> String {
```
Comment (jakepetroules) [inline Foundation/Process.swift]: Please remove the trailing semicolons from within this function

Comment (jmittert) [reply Foundation/Process.swift]: Oof, that's what I get for copying cpp code without thinking about it.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by jmittert** [comment]
Seems like something else failed?
```
<unknown>:0: error: module file '/tmp/org.llvm.clang.buildnode/ModuleCache/26WBVC9BAJTXE/_Builtin_stddef_max_align_t-2M3A8EWILE044.pcm' is out of date and needs to be rebuilt: signature mismatch
08:39:23 <unknown>:0: note: imported by module 'CoreFoundation' in '/tmp/org.llvm.clang.buildnode/ModuleCache/26WBVC9BAJTXE/CoreFoundation-2V85TRDDMO0NQ.pcm'
08:39:23 swiftc: /home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/llvm/tools/clang/lib/Serialization/InMemoryModuleCache.cpp:36: llvm::MemoryBuffer &clang::InMemoryModuleCache::addBuiltPCM(llvm::StringRef, std::unique_ptr<llvm::MemoryBuffer>): Assertion `!PCM.IsFinal && "Trying to override finalized PCM?"' failed.
08:39:23 Stack dump:
08:39:23 0.	<eof> parser at end of file
```

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**